src/services/PostService.ts

```diff
@@ -21,30 +21,56 @@ export class PostService {
     this.transactions = new TransactionsManager(entityManager);
   }
 
-  public async getAllPosts(): Promise<PostModel[]> {
+  public async getAllPosts(user: UserModel): Promise<PostModel[]> {
     return this.transactions.readOnly(async (transactionalEntityManager) => {
       const postRepository = Repositories.post(transactionalEntityManager);
-      // filter out posts from inactive users
-      return (await postRepository.getAllPosts()).filter((post) => post.user?.isActive);
+      const activeUserPosts = (await postRepository.getAllPosts()).filter((post) => post.user?.isActive);
+      const userRepository = Repositories.user(transactionalEntityManager);
+      const userWithBlockedInfo = await userRepository.getUserWithBlockedInfo(user.id)
+      const blockedUsers = userWithBlockedInfo?.blocking;
+      const unblockedPosts = activeUserPosts.filter((post) => {
+        if (blockedUsers) {
+          for (const blockedUser of blockedUsers) {
+            if (post.user?.id == blockedUser.id) return false;
+          }
+        }
+        return true;
+      });
+      return unblockedPosts; 
     });
   }
 
-  public async getPostById(params: UuidParam): Promise<PostModel> {
+  public async getPostById(user: UserModel, params: UuidParam): Promise<PostModel> {
     return this.transactions.readOnly(async (transactionalEntityManager) => {
       const postRepository = Repositories.post(transactionalEntityManager);
       const post = await postRepository.getPostById(params.id);
       if (!post) throw new NotFoundError('Post not found!');
       if (!post.user?.isActive) throw new NotFoundError('User is not active!');
+      const userRepository = Repositories.user(transactionalEntityManager);
+      const userWithBlockedInfo = await userRepository.getUserWithBlockedInfo(user.id)
+      const blockedUsers = userWithBlockedInfo?.blocking;
+      if (blockedUsers) {
+        for (const blockedUser of blockedUsers) {
+          if (post.user?.id == blockedUser.id) throw new ForbiddenError('User is blocked!');
+        }
+      }
```

Comment:
Same comment here as the one above, you can write an additional helper method that just throws an error whenever a post or array of posts contains a post from someone a user has blocked called something like `validatePostNotBlocked()`, and then just call this function everywhere you have this logic